### PR TITLE
Reduce torch eager dispatch overhead

### PR DESCRIPTION
**Pull request by MarcosAsh**
## Summary                                                                                            
   
  Reduces Keras[torch] eager dispatch overhead by avoiding unnecessary allocations and indirection in the hot path.                                             

  Addresses #22561.

  ## Changes

  - any_symbolic_tensors: iterate args directly instead of tree.flatten            
  - convert_to_tensor: fast path for already-correct torch tensors
  - Operation.__call__: inline error handling, skip wrapper creation per call                     
  - Layer.__call__: fast path for single-tensor eager inference (bypasses CallSpec, input validation, mask population, autocast scoping)                                                        
  - _set_mask_metadata/ mask population: skip tree ops for single tensors
  - assert_input_compatibility: skip tree.flatten for single-spec case                          
                                                            
  ## Benchmark                                                                                         
  
  [Colab notebook](https://colab.research.google.com/drive/1alwGHr8QhrVMbVDhoHad_3qXPaD5PVh_?usp=sharing) on T4
   GPU:                                                                                                 
                                                            
  | Metric | PyTorch | master | This PR | Improvement |
  |--------|---------|--------|---------|-------------|
  | add (256x256) | 9.0us | 51.5us (5.71x) | 16.3us (1.81x) | **3.2x faster** |                       
  | matmul (256x256) | 17.9us | 90.3us (5.04x) | 43.9us (2.45x) | **2.1x faster** |                   
  | softmax (64x1000) | 8.8us | 51.8us (5.91x) | 17.8us (1.95x) | **2.9x faster** |                   
  | CNN forward | 232.1us | 3099.7us (13.35x) | 1896.7us (8.50x) | **1.6x faster** |                    
  | Dense __call__ overhead | -- | 263.2us | 46.8us | **5.6x less overhead** |                        
                                                                                                        
  The remaining CNN gap includes Conv2D's NHWC/NCHW data format conversion overhead, which is a separate problem (#18457).

### CODE REVIEW

**Comment by MarcosAsh** [comment]
I think this was too large a PR so I am doing them in more digestible smaller PRs.
